cmd/ttn-lw-cli/commands/end_devices.go

```diff
@@ -331,10 +331,10 @@ var (
 
 			device.SetFields(res, "ids.dev_addr")
 			device.SetFields(res, append(append(nsPaths, asPaths...), jsPaths...)...)
-			if device.CreatedAt.IsZero() || (!res.CreatedAt.IsZero() && res.CreatedAt.Before(res.CreatedAt)) {
+			if ttnpb.StdTime(device.CreatedAt).IsZero() || (!ttnpb.StdTime(res.CreatedAt).IsZero() && ttnpb.StdTime(res.CreatedAt).Before(*ttnpb.StdTime(res.CreatedAt))) { // does this make sense?
```

Comment:
Instead of `ttnpb.StdTime(field).IsZero()` you can just `nil` check (`==nil`). Basically it's either `nil`, or it is safe to be turned into a `time.Time` using `ttnpb.StdTime`.

Same below.